boltdb/boltdb_record.go

```diff
@@ -0,0 +1,87 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package boltdb
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/timshannon/bolthold"
+)
+
+// StoredMetricRecord defines a Datastore entity that is used to store status information about an imported metric.
+type StoredMetricRecord struct {
+	Name        string
+	Query       string
+	LastUpdate  time.Time // last time we wrote any points to SD.
+	LastAttempt time.Time // last time we attempted an update.
+	LastStatus  string
+
+	// CounterStartTime is used to keep start timestamp for cumulative metrics.
+	CounterStartTime time.Time
+
+	storage *Manager
+}
+
+// Write metric data back to Datastore.
+func (m *StoredMetricRecord) write() error {
+	if err := m.storage.Store.Upsert(m.Name, m); err != nil {
+		return err
+	}
+	return nil
```

Comment:
Can this be simply `return m.storage.Store.Upsert(m.Name, m)`?

Reply:
Indeed it can, fixed!